master/test/integration/api/api_trials_test.go

```diff
@@ -134,6 +162,135 @@ func trialDetailAPITests(
 
 }
 
+func trialProfilerMetricsTests(
+	t *testing.T, creds context.Context, cl apiv1.DeterminedClient, db *db.PgDB,
+) {
+	experiment := testutils.ExperimentModel()
+	err := db.AddExperiment(experiment)
+	assert.NilError(t, err, "failed to insert experiment")
+
+	trial := testutils.TrialModel(experiment.ID, testutils.WithTrialState(model.ActiveState))
+	err = db.AddTrial(trial)
+	assert.NilError(t, err, "failed to insert trial")
+
+	ctx, _ := context.WithTimeout(creds, time.Minute)
+	tlCl, err := cl.GetTrialProfilerMetrics(ctx, &apiv1.GetTrialProfilerMetricsRequest{
+		Labels: &trialv1.TrialProfilerMetricLabels{
+			TrialId:    int32(trial.ID),
+			Name:       "gpu_util",
+			AgentId:    "brad's agent",
+			MetricType: trialv1.TrialProfilerMetricLabels_PROFILER_METRIC_TYPE_SYSTEM,
+		},
+	})
+	assert.NilError(t, err, "failed to initiate trial profiler metrics stream")
+
+	for i := 0; i < 10; i++ {
```

Comment:
This testing logic is substantial enough that I would suggest adding comments describing what we are testing for